### DISCUSSION TITLE
LVCMOS trigger signal with NIDAQ and ImSwitch

### DISCUSSION THREAD

**Discussion by hzarei4** [ImSwitch/ImSwitch#185] [Q&A]
Hello all,

I have a camera and is connected using the trigger port to the NIDAQ to take images. The camera supports the LVCMOS 3.3V triggering signal to take images. Is it possible to generate LVCMOS signal with a NIDAQ card "6251" and send it to the trigger port of the camera? Is it available with ImSwitch, too?

I already have tried the TTL scan signal generator of the ImSwitch but it did not work.

Thanks for your help.

**Comment by jacopoabramo** [comment]
Is it an existing manager or a new custom one?

**Reply by hzarei4** [reply]
I used an existing class called `class BetaTTLCycleDesigner`, but it did now work with the camera. The camera is a Hamamatsu camera.

**Reply by hzarei4** [reply]
Apparently, the camera only supports the LV (Low Voltage) signal rather than the TTL signal for the triggering.

**Reply by jacopoabramo** [reply]
Pay attention that if you feed a TTL signal to a LV signal pin you may risk damaging the instrument.
Did you change the settings of the Hamamatsu camera from the GUI to accept a trigger which a rising edge/high level signal on that specific line?

**Reply by hzarei4** [reply]
The Hamamatsu camera trigger option of the ImSwitch GUI only has 3 choices: 1. Internal trigger 2. Start time trigger 3. Frame trigger. Which one do you mean?

**Reply by jacopoabramo** [reply]
I did not use these seetings too much. From what I'm looking at the source code you should be using the "External 'frame-trigger'" option, but it does not specify which trigger pin is currently used. Could you write me the camera model?

**Reply by hzarei4** [reply]
Yes, I am using the External frame-trigger option. I tried to apply a 3.0V using the analog output (AO) of the NIDAQ to the trigger input pin of the camera and lower it to 0.0 so that it can catch both rising and falling edge signals to take an image but it did not work.

The camera which I am using is Hamamatsu ORCA-flash 2.8 (Model: C11440-10C).

**Reply by jacopoabramo** [reply]
Do you have an oscilloscope to verify that you're actually seeing 3.0 V on the AO channel? It may be that you're not genereting the signal at all.

**Reply by hzarei4** [reply]
That would be strange. Because there are two analog output channels that I am using. AO1 is for the lens and it is working with the applied signal. AO0 is for the trigger input of the camera.
But I will test the generated signal using a multimeter or oscilloscope.

**Reply by hzarei4** [reply]
Hi @jacopoabramo!
I tested the analog output AO0 of the NIDAQ card which is for the trigger input of the camera, using an oscilloscope. The voltage signal generator of the NIDAQ card is working correctly. 
One strange thing is that when I put the camera in the live view mode in the ImSwitch application, the trigger output of the camera does not generate any signal! Instead, it is supposed to generate a signal when the readout of a frame is done until the next frame exposure starts.
Have you experienced this problem before?

**Reply by jacopoabramo** [reply]
Hi @hzarei4 , sorry for the late reply. I can't say that I have, since I didn't use the Hamamatsu cameras with external triggers
